libdeno/binding.cc

```diff
@@ -208,17 +210,35 @@ void Print(const v8::FunctionCallbackInfo<v8::Value>& args) {
   stream << cstr << std::endl;
 }
 
-static v8::Local<v8::Uint8Array> ImportBuf(v8::Isolate* isolate, deno_buf buf) {
+static v8::Local<v8::Uint8Array> ImportBuf(Deno* d, deno_buf buf) {
   if (buf.alloc_ptr == nullptr) {
     // If alloc_ptr isn't set, we memcpy.
     // This is currently used for flatbuffers created in Rust.
-    auto ab = v8::ArrayBuffer::New(isolate, buf.data_len);
+
+    // To avoid excessively allocating new ArrayBuffers, we try to reuse a
+    // single global ArrayBuffer. The caveat is that users must extract data
+    // from it before the next tick. We only do this for ArrayBuffers less than
+    // 1024 bytes.
+    v8::Local<v8::ArrayBuffer> ab;
+    if (buf.data_len > GLOBAL_IMPORT_BUF_SIZE) {
+      // Simple case. We allocate a new ArrayBuffer for this.
```

Comment:
Now all we need to do to complete the picture is actually take of the deno_buf memory when the buffer is large.

Reply:
Yah. It'd be good to get a benchmark that is slow with the current setup. Maybe using readFile on a large file?